### ISSUE TITLE
meson build error on darwin

### ISSUE THREAD

**Issue by yanggeorge** [cisco/openh264#3414]
version: tags/v2.1.1

here is the meson-log.txt
```
Build started at 2021-08-28T21:41:17.368288
Main binary: /usr/local/opt/python@3.9/bin/python3.9
Build Options: 
Python system: Darwin
The Meson build system
Version: 0.59.1
Source dir: /Users/ym/me/c/gst-build/subprojects/openh264
Build dir: /Users/ym/me/c/gst-build/subprojects/openh264/ym
Build type: native build
Project name: openh264
Project version: 2.1.1
Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe -I/usr/local/opt/postgresql@12/include -L/usr/local/opt/postgresql@12/lib
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/sanitycheckc.exe
C compiler for the host machine: cc (clang 12.0.5 "Apple clang version 12.0.5 (clang-1205.0.22.11)")
C linker for the host machine: cc ld64 650.9
Sanity testing C++ compiler: c++
Is cross compiler: False.
Sanity check compiler command line: c++ sanitycheckcpp.cc -o sanitycheckcpp.exe -I/usr/local/opt/postgresql@12/include -L/usr/local/opt/postgresql@12/lib
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/sanitycheckcpp.exe
C++ compiler for the host machine: c++ (clang 12.0.5 "Apple clang version 12.0.5 (clang-1205.0.22.11)")
C++ linker for the host machine: c++ ld64 650.9
Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe -I/usr/local/opt/postgresql@12/include -L/usr/local/opt/postgresql@12/lib
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/sanitycheckc.exe
C compiler for the build machine: cc (clang 12.0.5 "Apple clang version 12.0.5 (clang-1205.0.22.11)")
C linker for the build machine: cc ld64 650.9
Sanity testing C++ compiler: c++
Is cross compiler: False.
Sanity check compiler command line: c++ sanitycheckcpp.cc -o sanitycheckcpp.exe -I/usr/local/opt/postgresql@12/include -L/usr/local/opt/postgresql@12/lib
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/sanitycheckcpp.exe
C++ compiler for the build machine: c++ (clang 12.0.5 "Apple clang version 12.0.5 (clang-1205.0.22.11)")
C++ linker for the build machine: c++ ld64 650.9
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Program nasm found: YES (/usr/local/bin/nasm)
Running compile:
Working directory:  /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmpzsconwy6
Command line:  c++ -I/usr/local/opt/postgresql@12/include /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmpzsconwy6/testfile.cpp -o /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmpzsconwy6/output.obj -c -O0 -fpermissive -Werror=unknown-warning-option -Werror=unused-command-line-argument -Werror=ignored-optimization-argument -Wnon-virtual-dtor -Wno-non-virtual-dtor 

Code:
 extern int i;
int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C++ supports arguments -Wno-non-virtual-dtor: YES 
Running compile:
Working directory:  /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmp0o7kanxk
Command line:  c++ -I/usr/local/opt/postgresql@12/include /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmp0o7kanxk/testfile.cpp -o /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmp0o7kanxk/output.obj -c -O0 -fpermissive -Werror=unknown-warning-option -Werror=unused-command-line-argument -Werror=ignored-optimization-argument -Wclass-memaccess -Wno-class-memaccess 

Code:
 extern int i;
int i;

Compiler stdout:
 
Compiler stderr:
 error: unknown warning option '-Wclass-memaccess'; did you mean '-Wclass-varargs'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wno-class-memaccess'; did you mean '-Wno-class-varargs'? [-Werror,-Wunknown-warning-option]

Compiler for C++ supports arguments -Wno-class-memaccess: NO 
Running compile:
Working directory:  /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmpoc8zc60p
Command line:  c++ -I/usr/local/opt/postgresql@12/include /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmpoc8zc60p/testfile.cpp -o /Users/ym/me/c/gst-build/subprojects/openh264/ym/meson-private/tmpoc8zc60p/output.obj -c -O0 -fpermissive -Werror=unknown-warning-option -Werror=unused-command-line-argument -Werror=ignored-optimization-argument -Wstrict-aliasing -Wno-strict-aliasing 

Code:
 extern int i;
int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C++ supports arguments -Wno-strict-aliasing: YES 
Run-time dependency threads found: YES

meson.build:86:2: ERROR: Problem encountered: FIXME: Unhandled system darwin

```

**Comment by GuangweiWang** [comment]
can you help to provide your command line? thanks

**Comment by huili2** [comment]
If no update, this will be closed.

**Comment by GuangweiWang** [comment]
you can use following command to build openh264 in Mac
`make`

**Comment by eli-schwartz** [comment]
This issue was fixed in commit a396835831f9a8468f322cab2f584e133cb979eb ("add support for android, ios, macos") and released in tag v2.2.0
